src/System.Text.Json/tests/Utf8JsonReaderTests.TextEquals.cs

```diff
@@ -0,0 +1,447 @@
+                    }
+                }
+            }
+
+            Assert.Equal(expectedFound, found);
+        }
+
+        [Theory]
+        [InlineData("{\"name\": \"John\"}", "John", true)]
+        [InlineData("{\"name\": \"John\"}", "Johna", false)]
+        [InlineData("{\"name\": \"John\"}", "Joh\\u006e", false)]
+        [InlineData("{\"name\": \"John\"}", "", false)]
+        [InlineData("{\"name\": \"\"}", "John", false)]
+        [InlineData("{\"name\": \"\"}", "Joh\\u006e", false)]
+        [InlineData("{\"name\": \"\"}", "", true)]
+        [InlineData("{\"name\": \"Joh\\u006e\"}", "John", true)]
+        [InlineData("{\"name\": \"Joh\\u006e\"}", "Johna", false)]
+        [InlineData("{\"name\": \"Joh\\u006e\"}", "Joh\\u006e", false)]
+        [InlineData("{\"name\": \"Joh\\u006e\"}", "", false)]
+        public static void TestTextEqualsValue(string jsonString, string lookUpString, bool expectedFound)
+        {
+            byte[] lookup = Encoding.UTF8.GetBytes(lookUpString);
+            byte[] utf8Data = Encoding.UTF8.GetBytes(jsonString);
+            bool found = false;
+
+            var json = new Utf8JsonReader(utf8Data, isFinalBlock: true, state: default);
+            while (json.Read())
+            {
+                if (json.TokenType == JsonTokenType.String)
+                {
+                    if (json.TextEquals(lookup) && json.TextEquals(lookUpString.AsSpan()))
+                    {
+                        found = true;
+                    }
+                }
+            }
+
+            Assert.Equal(expectedFound, found);
+
+            ReadOnlySequence<byte> sequence = JsonTestHelper.GetSequence(utf8Data, 1);
+            found = false;
+
+            json = new Utf8JsonReader(sequence, isFinalBlock: true, state: default);
+            while (json.Read())
+            {
+                if (json.TokenType == JsonTokenType.String)
+                {
+                    if (json.TextEquals(lookup) && json.TextEquals(lookUpString.AsSpan()))
+                    {
+                        found = true;
+                    }
+                }
+            }
+
+            Assert.Equal(expectedFound, found);
+        }
+
+        [Fact]
+        public static void TestTextEqualsLargeMatch()
+        {
+            var jsonChars = new char[320];  // Some value larger than 256 (stack threshold)
+            jsonChars.AsSpan().Fill('a');
+            byte[] lookup = Encoding.UTF8.GetBytes(jsonChars);
+            ReadOnlySpan<byte> lookupSpan = lookup.AsSpan(0, lookup.Length - 5);   // remove extra characters that were replaced by escaped bytes
+            Span<char> lookupChars = new char[jsonChars.Length];
+            jsonChars.CopyTo(lookupChars);
+            lookupChars = lookupChars.Slice(0, lookupChars.Length - 5);
+
+            ReadOnlySpan<char> unEscapedA = new char[6] { '\\', 'u', '0', '0', '6', '1' };
+
+            for (int i = 0; i < jsonChars.Length - 5; i++)
```

Comment:
why 5?

Reply:
`unEscapedA.Length` - 1 = 6 - 1 = 5
Replacing 'a' with '\u0061', so net change of 5.